pkg/providers/nutanix/provider.go

```diff
@@ -496,6 +496,11 @@ func (p *Provider) UpdateKubeConfig(content *[]byte, clusterName string) error {
 	return nil
 }
 
+// HardwareSpec returns the hardware yaml containing configurations for the hardware in the cluster.
+func (p *Provider) HardwareSpec() []byte {
```

Comment:
`HardwareSpec()` doesn't make much sense for any providers besides tinkerbell. Is there a different way we can get the hardware spec without adding this func to the provider interface?

Reply:
I figured it would be an okay approach, since we've done something similar with `UpdateKubeConfig`.

This is how I planned on using it. in the CLI flow:
```
func (s *InstallGitOpsManagerTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
...
	err := commandContext.GitOpsManager.InstallGitOps(..., ..., ..., ..., ..., commandContext.Provider.HardwareSpec())
...
}
```
Another approach is to have it only on the Tinkerbell Provider type and then use a type assertion to access it from the CLI flow when needed if that's preferred?